src/libsyntax/parse/parser.rs

```diff
@@ -8587,6 +8588,17 @@ impl<'a> Parser<'a> {
             }
         }
     }
+
+    /// Recover from `pub` keyword in places where it seems _reasonable_ but isn't valid.
+    fn eat_bad_pub(&mut self) {
+        if self.token.is_keyword(keywords::Pub) {
+            self.bump();
```

Comment:
It probably makes sense to parse the whole visibility (`pub(...)`) and discard it rather than discard the `pub` only.